File: pset6/caesar.py
import sys

def main():
    if len(sys.argv)!=2 or not sys.argv[1].isdigit():
        print("Wrong number of arguments. Please try again.")
        return 1
    
    k = int(sys.argv[1])
    
    p = input("plaintext: ")
    print("ciphertext: ", end="")
    for i in range(len(p)):
        if p[i].isupper():
            print(chr(ord('A') + (ord(p[i]) - ord('A') + k) % 26 ), end="")
        elif p[i].islower():
            print(chr(ord('a') + (ord(p[i]) - ord('a') + k) % 26 ), end="")
        else:
            print(p[i], end="")
    
    print("")
    
    return 0
    
    
if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])  
{
    if (argc != 2 && !isdigit(argv[1]))
    {
        printf ("Wrong number of arguments. Please try again.\n");
        return 1;
    }

    int k = atoi(argv[1]);

    printf("plaintext: ");
    string p = GetString();
    
    printf("ciphertext: ");
    for (int i = 0, n = strlen(p); i < n; i++)
    {            
        if (isupper(p[i]))
            printf("%c", 'A' + (p[i] - 'A' + k) % 26);
        else if (islower(p[i]))
            printf("%c", 'a' + (p[i] - 'a' + k) % 26);
        else
            printf("%c", p[i]);
    }

    printf("\n");

    return 0;
}    